PythonServiceTest.batchStage_mapUsingPython [HZ-1756]

The `PythonServiceTest.batchStage_mapUsingPython` failed on PR builder on `master` branch
https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/13859/

<details><summary>Stacktrace:</summary>

```
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 1
	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
	at com.hazelcast.jet.Job.join(Job.java:78)
	at com.hazelcast.jet.python.PythonServiceTest.batchStage_mapUsingPython(PythonServiceTest.java:105)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
Caused by: com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 1
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:114)
	at com.hazelcast.jet.python.PythonService.createContextWithRetry(PythonService.java:109)
	at com.hazelcast.jet.python.PythonService.lambda$factory$9a0f8f23$1(PythonService.java:89)
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:49)
	at com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.init(ProcessorSupplierWithService.java:59)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.lambda$null$9c5eb802$1(ExecutionPlan.java:456)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:532)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.lambda$initProcSuppliers$10beb6ff$1(ExecutionPlan.java:455)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 1
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:105)
	... 16 more
```

</details>

<details><summary>Standard output:</summary>

```
14:54:46,596 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] + python3 -m venv jet_to_python_env --system-site-packages
14:54:50,761 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] + source jet_to_python_env/bin/activate
14:54:50,763 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ deactivate nondestructive
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' -n '' ']'
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' -n '' ']'
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' -n /bin/bash -o -n '' ']'
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ hash -r
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' -n '' ']'
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ unset VIRTUAL_ENV
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' '!' nondestructive = nondestructive ']'
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ VIRTUAL_ENV=/home/jenkins/tmp/jet-compassionate_ishizaka-08d6-de26-5040-0001-_home_jenkins_tmp_jet-test-temp36122603625161707389/jet_to_python_env
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ export VIRTUAL_ENV
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ _OLD_VIRTUAL_PATH=/usr/share/jdks/oracle-8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ PATH=/home/jenkins/tmp/jet-compassionate_ishizaka-08d6-de26-5040-0001-_home_jenkins_tmp_jet-test-temp36122603625161707389/jet_to_python_env/bin:/usr/share/jdks/oracle-8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
14:54:50,764 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ export PATH
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' -n '' ']'
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' -z '' ']'
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ _OLD_VIRTUAL_PS1=
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' 'x(jet_to_python_env) ' '!=' x ']'
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ PS1='(jet_to_python_env) '
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ export PS1
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ '[' -n /bin/bash -o -n '' ']'
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ++ hash -r
14:54:50,765 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] + python3 -m pip install --upgrade pip
14:54:50,769 DEBUG || - [JobRepository] hz.compassionate_ishizaka.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 28ms
14:54:53,483 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Collecting pip
14:54:53,496 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT]   Using cached pip-21.3.1-py3-none-any.whl (1.7 MB)
14:54:53,689 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Installing collected packages: pip
14:54:53,690 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT]   Attempting uninstall: pip
14:54:53,690 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT]     Found existing installation: pip 20.0.2
14:54:53,828 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT]     Uninstalling pip-20.0.2:
14:54:53,834 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT]       Successfully uninstalled pip-20.0.2
14:54:55,282 DEBUG || - [MasterJobContext] hz.compassionate_ishizaka.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Not scaling up job '08d6-de26-5040-0001', execution 08d6-de26-5041-0001: not running or already running on all members
14:54:55,698 DEBUG || - [JobRepository] hz.compassionate_ishizaka.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 23ms
14:54:55,844 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Successfully installed pip-21.3.1
14:54:55,975 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] + python3 -m pip install protobuf==3.19.4 grpcio==1.48.0
14:54:57,176 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ERROR: Could not find a version that satisfies the requirement protobuf==3.19.4 (from versions: 2.0.0b0, 2.0.3, 2.3.0, 2.4.1, 2.5.0, 2.6.0, 2.6.1, 3.0.0a2, 3.0.0a3, 3.0.0b1, 3.0.0b1.post1, 3.0.0b1.post2, 3.0.0b2, 3.0.0b2.post1, 3.0.0b2.post2, 3.0.0b3, 3.0.0b4, 3.0.0, 3.1.0, 3.1.0.post1, 3.2.0rc1, 3.2.0rc1.post1, 3.2.0rc2, 3.2.0, 3.3.0, 3.4.0, 3.5.0.post1, 3.5.1, 3.5.2, 3.5.2.post1, 3.6.0, 3.6.1, 3.7.0rc2, 3.7.0rc3, 3.7.0, 3.7.1, 3.8.0rc1, 3.8.0, 3.9.0rc1, 3.9.0, 3.9.1, 3.9.2, 3.10.0rc1, 3.10.0, 3.11.0rc1, 3.11.0rc2, 3.11.0, 3.11.1, 3.11.2, 3.11.3, 3.12.0rc1, 3.12.0rc2, 3.12.0, 3.12.1, 3.12.2, 3.12.4, 3.13.0rc3, 3.13.0, 3.14.0rc1, 3.14.0rc2, 3.14.0rc3, 3.14.0, 3.15.0rc1, 3.15.0rc2, 3.15.0, 3.15.1, 3.15.2, 3.15.3, 3.15.4, 3.15.5, 3.15.6, 3.15.7, 3.15.8, 3.16.0rc1, 3.16.0rc2, 3.16.0, 3.17.0rc1, 3.17.0rc2, 3.17.0, 3.17.1, 3.17.2, 3.17.3, 3.18.0rc1, 3.18.0rc2, 3.18.0, 3.18.1, 3.19.0rc1, 3.19.0rc2, 3.19.0, 3.19.1, 4.0.0rc1, 4.0.0rc2)
14:54:57,177 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@7b093eac - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] ERROR: No matching distribution found for protobuf==3.19.4
14:54:57,701 DEBUG || - [python] python-init-logger_java.lang.UNIXProcess@144a276c - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] + python3 -m venv jet_to_python_env --system-site-packages
```

</details>

--- a/extensions/python/src/main/resources/jet_to_python_init.sh
+++ b/extensions/python/src/main/resources/jet_to_python_init.sh
@@ -5,8 +5,10 @@ set -e -x
 
 python3 -m venv jet_to_python_env --system-site-packages
 source jet_to_python_env/bin/activate
-python3 -m pip install --upgrade pip
-python3 -m pip install protobuf==@protobuf.version@ grpcio==@grpc.version@
+
+flock --exclusive ~/.jet-pip.lock python3 -m pip install --upgrade pip
+flock --exclusive ~/.jet-pip.lock python3 -m pip install protobuf==@protobuf.version@ grpcio==@grpc.version@
+
 [[ -f requirements.txt ]] && python3 -m pip install -r requirements.txt
 source ./jet_to_python_params.sh
 if [[ -f init.sh ]]; then

